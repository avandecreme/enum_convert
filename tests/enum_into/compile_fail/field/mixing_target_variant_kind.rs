use enum_convert::EnumInto;

#[derive(EnumInto)]
#[enum_into(Target)]
enum Source {
    #[enum_into(Target::Struct)]
    Struct {
        #[enum_into(Target::Struct.0)] // .0 Does not make sense for a Struct target
        a: i32,
        #[enum_into(Target::Struct.b)]
        b: i32,
    },
}

enum Target {
    Struct { a: i32, b: i32 },
}

fn main() {}
