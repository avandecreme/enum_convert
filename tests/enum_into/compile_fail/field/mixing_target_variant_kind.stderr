error: Unexpected mapping to positional field for struct variant
 --> tests/enum_into/compile_fail/field/mixing_target_variant_kind.rs:8:36
  |
8 |         #[enum_into(Target::Struct.0)] // .0 Does not make sense for a Struct target
  |                                    ^
